docs/site/Validation-REST-layer.md

```diff
@@ -175,3 +175,76 @@ of 10 digits separated by `-` after the 3rd and 6th digits.
 ```
 
 {% include tip.html content="RegExp can be converted into a string with `.source` to avoid escaping backslashes" %}
+
+## Customize validation errors
+
+Since the error is being caught at the REST layer, the simplest way to customize
+the errors is to customize the
+[sequnce](https://loopback.io/doc/en/lb4/Sequence.html). It exists in all
+LoopBack applications scaffolded by using the `lb4` command and can be found in
+`src/sequence.ts`.
+
+Let's take a closer look at how to customize the error. A few things to note in
+the below code snippet:
+
+1. inject RestBindings.ERROR_WRITER_OPTIONS for logging error
+2. customize error for particular endpoints
+3. create a new error with customized properties
+4. log the error using RestBindings.SequenceActions.LOG_ERROR
+
+{% include code-caption.html content="/src/sequence.ts" %}
+
+```ts
+// 1. inject RestBindings.ERROR_WRITER_OPTIONS for logging error
+constructor(
+    @inject(SequenceActions.FIND_ROUTE) protected findRoute: FindRoute,
+    @inject(SequenceActions.PARSE_PARAMS) protected parseParams: ParseParams,
+    @inject(SequenceActions.INVOKE_METHOD) protected invoke: InvokeMethod,
+    @inject(SequenceActions.SEND) public send: Send,
+    @inject(SequenceActions.REJECT) public reject: Reject,
+    @inject(RestBindings.ERROR_WRITER_OPTIONS, {optional: true})
+    protected errorWriterOptions?: ErrorWriterOptions,
+  ) {}
+
+async handle(context: RequestContext) {
+    try {
+      //...
+    } catch (err) {
+      const httpError = <HttpErrors.HttpError>err;
```

Comment:
The catch block is very long, please extract it into a new method.

```ts
class MySequence /*...*/ {
  // ...
 async handle(context: RequestContext) {
     try {
       //...
     } catch (err) {
        this.handleError(err);
     }
  }

  handleError(err: Error) {
    const httpError = <HttpErrors.HttpError>err;
    // ...

    // Otherwise fall back to the default error handler
    this.reject(context, err); 
  }
}
```